lib/ansible/module_utils/basic.py

```diff
@@ -1036,6 +1036,10 @@ def _symbolic_mode_to_octal(cls, path_stat, symbolic_mode):
 
         new_mode = stat.S_IMODE(path_stat.st_mode)
 
+        if not stat.S_ISDIR(path_stat.st_mode):
+            # chmod -X removes executable bit from files, +X does not add it
+            symbolic_mode = symbolic_mode.replace('+X', '')
```

Comment:
This needs to strip out `X` too, as long as it is not preceded by `-`.

It was also suggested that +X just be an error for files, but since this code is used recursively for directories, it would be easier to do that in the module.